### PR TITLE
Drone hatches

### PR DESCRIPTION
**Pull request by ike709**
:cl: ike709
add: Ported drone hatches from Aurora. Now maintenance drones shouldn't let people into places anymore.
/:cl:

Technically, this comes from the FTL13 pull request that ports these from Aurora. I did change it a bit though, like removing the hatches from mineral doors.

I forgot to record audio, but here's some videos anyways.
https://www.dropbox.com/s/d4li9ksrqgnpwdg/2017-10-15-0148-49.mp4?dl=0
https://www.dropbox.com/s/xeg5l3vuq4avu8l/2017-10-15-0149-37.mp4?dl=0

I also added the GitHub Atom build file to the gitignore to make my life easier.

### CODE REVIEW

**Comment by ike709** [comment]
It occurs to me that this will probably get closed due to the freeze. Whatever, just reopen it after.

**Comment by Bawhoppen** [comment]
While I understand the idea behind this, I disagree with the fact that drones should be completely detached from reality and affect the round in no subversive ways at all.

**Comment by MoreRobustThanYou** [comment]
@Bawhoppen They DO affect the round, by repairing or improving machines and stuff

**Comment by Bawhoppen** [comment]
Did you see how I included the word SUBVERSIVE

**Comment by PKPenguin321** [comment]
Oh that's so neat

**Comment by davethwave** [comment]
The opening and closing of the doors needs spriting, they just kinda disapear when people open.

**Comment by ACCount12** [comment]
Looks way too ugly for my taste. Adding this kind of eye sore just for drones? No.

**Comment by ike709** [comment]
\>implying sprites can't be changed

**Comment by ACCount12** [comment]
This PR is 20% feature 80% eyesore sprites. Maybe better wait for the good ones?

**Comment by praisenarsie** [comment]
uhhhhhh drones can use the vents/scrubbers AND the plastic flaps.

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -98,6 +98,15 @@
...
 	var/static/list/airlock_overlays = list()
 
+	var/has_hatch = TRUE //If TRUE, this door has hatches, and certain small creatures can move through them without opening the door
+	var/hatchstate = 0 //0: closed, 1: open
+	var/hatch_offset_x = 0
+	var/hatch_offset_y = 0
+	var/hatch_colour = "#7d7d7d"
+	var/hatch_open_sound = 'sound/machines/hatch_open.ogg'
+	var/hatch_close_sound = 'sound/machines/hatch_close.ogg'
+	var/image/hatch_image
```
Comment (RemieRichards) [inline code/game/machinery/doors/airlock.dm]: make this a mutable appearance

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -152,6 +164,29 @@
 	. = ..()
 	AddComponent(/datum/component/rad_insulation, RAD_MEDIUM_INSULATION)
 
+/obj/machinery/door/airlock/proc/setup_hatch()
+	hatch_image = image('icons/obj/doors/hatches.dmi', src, "hatch_closed", layer+0.01)
```
Comment (RemieRichards) [inline code/game/machinery/doors/airlock.dm]: >make this a mutable appearance

**Comment by Dorsisdwarf** [comment]
This doesn't look great and I don't like the idea.

Also if you're doing this, at least dont be so incredibly lazy as to not bother adding the hatch sprites to opening and open doors.

**Comment by KorPhaeron** [comment]
Freeze

**Comment by ike709** [comment]
I'll fix it after the freeze is over.

@Dorsisdwarf That's not laziness, that's intentional. I thought it looked kind of odd having the drone hatch there when the door was opened, so instead I just let it look like the hatch opens when the door does. Instead of insulting me, please kindly word it as a suggestion.

**Comment by ike709** [comment]
@praisenarsie It takes time to enter the vent, and ventcrawling is a bit annoying to navigate. I never use it as a drone.

**Pull request closed without merging**